content/post/000-ipfs-in-web-browsers.md

```diff
@@ -0,0 +1,74 @@
+
+Decentralization feels slow in providing user-centric approaches to the internet's biggest problems. Right now true user agency is often too technical, too fragile and too time-consuming to be the default choice.
+
+But we're making headway. And we'd like to share some collaborations the IPFS project has had in the works for a while, which bring us just a few steps closer to making unmediated access to information *just work*... by solving that "last mile" problem and *integrating IPFS directly into web browsers*.
+
+## Stages
+
+The path to a truly decentralized web is a long one. For over 30 years the browser has been a *client* - but a foundational concept in P2P systems is that a participant is both a client *and a server*. Web browser vendors and web standards bodies have not designed for this architectural shift, so we're breaking it down into steps.
+
+Initially we're aiming to ease access to the IPFS HTTP gateways - creating a bridge between the billions of people on the centralized web and content on the distributed web. We expect some experimentation in the next phase - adjusting node behavior based on browser APIs, device capabilities, bandwidth availability and power consumption - growing adoption with our browser partners as we learn, and building common understanding at standards bodies. Finally we aim to have multiple browsers implementing IPFS natively and interoperably.
+
+(INSERT STAGES DIAGRAM)
+
+## Firefox
+
+> Perhaps it makes sense to mention Firefox as a full recap-like section here, afaik we did not blog about libdweb at all yet. 
+> Idea: start with mentioning we've been working with Mozilla on initial APIs since 2018, mention [experimental protocol handler](https://github.com/ipfs-shipyard/ipfs-companion/pull/533) and [local discovery + TCP transport](https://github.com/ipfs-shipyard/ipfs-companion/pull/553) and include video demos from 2018: user-facing handler https://www.youtube.com/watch?v=fS8pLOQdOoM + optionally https://www.youtube.com/watch?v=FRzyWUXIyeo then end this section with hopes to see those APIs start landing in Nightly as a soft pressure by shining some light on this. 
+> Thoughts? [name=lidel]
+
+## Brave
+
+Brave has tended to live up to its name since the beginning - riling up the browser, publisher and crypto worlds. [Since 2018](https://github.com/brave/brave-browser/issues/819) we've been working with Brave on IPFS support in some form, and the results of that work are now emerging.
+
+Today you can open [brave://settings/extensions](brave://settings/extensions) in Brave beta or nightly and see a toggle for enabling IPFS Companion.
+
+![Enabling IPFS Companion in Brave](https://file.globalupload.io/ZVxjmPZ48E.gif)
+
+Removing this install step *really matters* for adoption. While technical users are familiar with browser extensions, it's less comfortable for non-technical users.
```

Comment:
aren't chrome extensions also more complicated for Brave users?

Reply:
Hm, not that I know of. Chrome app store "just works" for Brave. Same source code internally, so shouldn't be any compat issues.

Reply:
Ah - I guess I'm just thinking of the complexity of going to the "chrome webstore" to manage extensions in Brave instead of brave://extensions (the equivalent of what you'd do in Chrome). That's a little unintuitive for a brave user - and having companion directly integrated into the brave extensions page definitely improves discoverability!!

Reply:
gotcha - yeah that's def some cognitive overhead for Brave users, added a bit about it

Reply:
fixed